Framework/Algorithms/test/ReflectometryBeamStatisticsTest.h

```diff
@@ -0,0 +1,366 @@
+        sum += angd[i] * pow<2>(angd_cen - static_cast<double>(i));
+      }
+      return sum;
+    }();
+    return 2. * std::sqrt(2. * std::log(2.)) * PIXEL_SIZE *
+           std::sqrt(tt / total_angd);
+  }
+
+  static double s3_fwhm(const double l2) {
+    // This function comes from COSMOS.
+    return 0.68 * SLIT2_SIZE / (SLIT2_DIST + l2);
+  }
+
+  static double err_ray(const double l2, const double angle_bragg,
+                        const std::string &sumType, const bool polarized,
+                        const double om_fwhm) {
+    // This function comes from COSMOS.
+    using namespace boost::math;
+    double err_ray1;
+    if (sumType == "SumInQ") {
+      if (om_fwhm > 0) {
+        if (S2_FWHM >= 2 * om_fwhm) {
+          err_ray1 = std::sqrt(pow<2>(DET_RESOLUTION / l2) +
+                               pow<2>(s3_fwhm(l2)) + pow<2>(om_fwhm)) /
+                     angle_bragg;
+        } else {
+          err_ray1 = std::sqrt(pow<2>(DET_RESOLUTION / (2. * l2)) +
+                               pow<2>(s3_fwhm(l2)) + pow<2>(S2_FWHM)) /
+                     angle_bragg;
+        }
+      } else {
+        if (S2_FWHM > DET_RESOLUTION / l2) {
+          err_ray1 =
+              std::sqrt(pow<2>(DET_RESOLUTION / l2) + pow<2>(s3_fwhm(l2))) /
+              angle_bragg;
+        } else {
+          err_ray1 = std::sqrt(pow<2>(da()) + pow<2>(DET_RESOLUTION / l2)) /
+                     angle_bragg;
+        }
+      }
+    } else {
+      if (polarized) {
+        err_ray1 = std::sqrt(pow<2>(da())) / angle_bragg;
+      } else {
+        err_ray1 = std::sqrt(pow<2>(da()) + pow<2>(om_fwhm)) / angle_bragg;
+      }
+    }
+    const auto err_ray_temp =
+        0.68 *
+        std::sqrt((pow<2>(PIXEL_SIZE) + pow<2>(SLIT2_SIZE)) / pow<2>(l2)) /
+        angle_bragg;
+    return std::min(err_ray1, err_ray_temp);
+  }
+
+  static double da() {
+    // This function comes from COSMOS.
+    using namespace boost::math;
+    return 0.68 * std::sqrt((pow<2>(SLIT1_SIZE) + pow<2>(SLIT2_SIZE)) /
+                            pow<2>(INTERSLIT));
+  }
+
+  static double om_fwhm(const double l2, const double dirl2,
+                        const double dirs2w, const double dirs3w,
+                        const double det_fwhm, const double detdb_fwhm) {
+    // This function comes from COSMOS.
+    using namespace boost::math;
+    const double sdr = SLIT2_DIST + l2;
+    const double ratio = SLIT2_SIZE / SLIT1_SIZE;
+    const double vs = sdr + (ratio * INTERSLIT) / (1 + ratio);
+    const double da_det = std::sqrt(pow<2>(da() * vs) + pow<2>(DET_RESOLUTION));
+    double om_fwhm{0};
+    if (std::abs(SLIT1_SIZE - dirs2w) >= 0.00004 ||
+        std::abs(SLIT2_SIZE - dirs3w) >= 0.00004) {
+      if ((det_fwhm - da_det) >= 0.) {
+        if (std::sqrt(pow<2>(det_fwhm) - pow<2>(da_det)) >= PIXEL_SIZE) {
+          om_fwhm = 0.5 * std::sqrt(pow<2>(det_fwhm) - pow<2>(da_det)) / dirl2;
+        } else {
+          om_fwhm = 0;
```

Comment:
0.

Reply:
Do you need to explicitly set om_fwhm again to 0.? It is already initialized by 0.

Reply:
The function is straight from COSMOS. In this form it was easier to compare it to the original code for validation but I think we're safe now to remove the quirks.